api/v1alpha1/customnbimage_types.go

```diff
@@ -82,6 +82,9 @@ type BuildTypeSpec struct {
 	// Repository is the URL of the git repository, used for building
 	// +optional
 	Repository string `json:"repository,omitempty"`
+	// GitRef is the git reference within the Repository to use for building (e.g. "main")
+	// +optional
+	GitRef string `json:"gitRef,omitempty"`
```

Comment:
how is this different from `Repository` above?

Reply:
It is a reference within the repository, e.g. a branch name (`main`), a tag, a specific commit where to build from

Reply:
oh ja :/